cargo-dist/src/cargo_build.rs

```diff
@@ -227,7 +227,11 @@ pub fn build_cargo_target(dist_graph: &DistGraph, target: &CargoBuildStep) -> Re
         FastMap::<String, FastMap<String, Vec<(Utf8PathBuf, Utf8PathBuf)>>>::new();
     for &binary_idx in &target.expected_binaries {
         let binary = &dist_graph.binary(binary_idx);
-        let package_id = binary.pkg_id.to_string();
+        let package_id = if let Some(id) = &binary.pkg_id {
+            id.to_string()
+        } else {
+            binary.name.to_owned()
```

Comment:
wait will this actually work? "package_id" is very specifically a special id we need to use to communicate with rustc/cargo correctly, i believe. it can't be substituted

Reply:
We additionally use it as the key to the hash map of all binaries that exist. While I'll be factoring a bunch of cargo-specific stuff out, that piece of logic is independent of what the underlying builder is.

Reply:
Updated this to a simpler `expect()` in the place it's used, which should no longer overlap with non-cargo usage.